### PR TITLE
Remove duplication of ConvertToErrorEvents

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #7436

This PR moves the error conversion into the internal directory, and includes the code improvement suggested by @bznein 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by bznein — APPROVED** [review]
LGTM

File: modules/core/04-channel/v2/keeper/msg_server.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ func (k *Keeper) RecvPacket(ctx context.Context, msg *channeltypesv2.MsgRecvPack
 			writeFn()
 		} else {
 			// Modify events in cached context to reflect unsuccessful acknowledgement
-			sdkCtx.EventManager().EmitEvents(convertToErrorEvents(cacheCtx.EventManager().Events()))
+			sdkCtx.EventManager().EmitEvents(internalerrors.ConvertToErrorEvents(cacheCtx.EventManager().Events()))
```
Comment (damiannolan) [inline modules/core/04-channel/v2/keeper/msg_server.go:130]: think this is duplicate import, see `ibcerrors`

Comment (damiannolan) [reply modules/core/04-channel/v2/keeper/msg_server.go:130]: oh wait, its not duplicate... should we just put it in `ibcerrors` none the less?

Comment (chatton) [reply modules/core/04-channel/v2/keeper/msg_server.go:130]: I guess we can do that yeah, no harm

Comment (chatton) [reply modules/core/04-channel/v2/keeper/msg_server.go:130]: oh, ibcerrors is exported, it's not under internal.

**Pull request merged**